### ISSUE TITLE
entering a string doesn't trigger the similar bugs

### ISSUE THREAD

**Issue by karlcow** [webcompat/webcompat.com#1999]
1. Go to https://staging.webcompat.com/issues/new
2. Log in
3. Type facebook in Site URL field.

Nothing happens.

**Comment by miketaylr** [comment]
I also experienced this...which prompted my email for STR.

**Comment by MDTsai** [comment]
It's expected, maybe? I make an URL object and get the hostname [here](https://github.com/webcompat/webcompat.com/blob/b763a4cb2f118959704762d5c57dd5e00d91264a/webcompat/static/js/lib/bugform.js#L234)

facebook or facebook.com and even www.facebook.com is not a valid URL. We can use URL to search if it's invalid URL, good idea?

**Comment by zoepage** [comment]
> It's expected, maybe? 

I agree. As it's the URL input field, we expect a valid URL for the dropdown to function.

> Nothing here is related to https://soundcloud.com/ that I get from typing the full domain name. https://github.com/webcompat/webcompat.com/issues/2000#issue-285885503

That is correct as the URL type stripes away the protocol. We use the type as we want to make sure to have a valid URL to search for. 

So, I'd say we need to decide on what we want to look for. Keyword? URL? And adapt the functionality.

**Comment by karlcow** [comment]
The search is happening on the full length of the title currently as said in #2000 

![capture d ecran 2018-01-10 a 13 32 52](https://user-images.githubusercontent.com/505230/34755944-dc047768-f60a-11e7-9ac5-9110bfb95034.png)

Suggestions:

1. The search kicks in without having to enter http:// or https://
2. One of these two:
  1. search_string is targeting only the domain name part of the title. It should not try to match anything else.
  2. search_string is targeting the full title as of now.
3. There is a need to explain that this will open a new tab for checking if it's about the same bug.

Then I was wondering: Would I use it? What do I need?

* As a user when I report bug, I use the "…" (Report Site Issue)  
  it triggers the page with the full already filed bug. ![capture d ecran 2018-01-10 a 13 41 22](https://user-images.githubusercontent.com/505230/34756137-1b179e48-f60c-11e7-9e3e-2ed4555b759a.png)

I'm unlikely to click in the URL part. It doesn't give me any information on similar bugs.

What do I need?

Maybe it's the hardest part. There are often things we think we need. 

The searchstring on the full title is interesting but has nothing to do with the URL. And probably should be designed in a different way. There's a feature in Apple addressbook that I like. "Search for duplicates" When it finds to card close enough, it proposes to merge them.

What I could imagine is that there could be a tool on webcompat.com which looks for similarities in between bugs learning for certain features, texts, keywords. Probably a bit AI-ish stuff. But this could probably be something interesting for the research team, or an intern. It could also probably be extended to bugzilla.

**Comment by MDTsai** [comment]
Back to beginning, what I (or Ola) want to do, is to improve the bug report form quality, to reduce number of duplicated issue. In #1826 we already restrict anonymous user to use this feature in my prototype. But even if it's a registered user, direct him to an existing bug seems not helpful now? As Karl's comment, it's because we don't show what they want to know. A better one might be title + description, with just fit title and description, does this make sense? So we can image that move to our end point is necessary, using github API is not enough.

Tool to find similar bug is also cool, like e-commerce recommendation :)

**Comment by MDTsai** [comment]
For both URL and description notification, I think a better search format is:
```Number + URL + Description```
In URL field, we can match same URL, in description field, match similar description/problem type/summary. Same format with same experience, but different match pattern.

**Comment by MDTsai** [comment]
I would like to change owner to @zoepage , please take care of this feature :)

**Comment by zoepage** [comment]
+1 @mdtsai
